orion-ui/tests/utilities/useToggle.ts

```diff
@@ -0,0 +1,55 @@
+import { Locator, Page } from '@playwright/test'
+import { PAGE } from './fixtures'
+
+export type UseToggle = {
+  toggle: Locator,
+  getState: () => Promise<boolean | undefined>,
+  setState: (state: boolean) => Promise<void>,
+}
+
+export function useToggle(page: Page | Locator = PAGE): UseToggle {
+  return {
+    toggle: getToggle(page),
+    getState: async (toggle: Locator = getToggle(page)) => {
+      const count = await toggle.count()
+      console.log({ count })
+      const promises: Promise<boolean>[] = []
+
+      for (let i = 0; i < count; ++i) {
+        promises.push(getToggleState(toggle.nth(i)))
+      }
+
+      const states = await Promise.all(promises)
+      if (new Set(states).size === 1) {
+        return states[0]
+      }
+    },
+    setState: async (state, toggle: Locator = getToggle(page)) => {
+      const count = await toggle.count()
+      const promises: Promise<void>[] = []
+
+      for (let i = 0; i < count; ++i) {
+        promises.push(setToggleState(toggle.nth(i), state))
+      }
+
+      await Promise.all(promises)
+    },
+  }
+}
+
+function getToggle(page: Page | Locator): Locator {
+  return page.locator('.p-toggle')
+}
+
+async function setToggleState(toggle: Locator, state: boolean): Promise<void> {
+  const current = await getToggleState(toggle)
+
+  if (current !== state) {
+    await toggle.click()
+    await toggle.locator(`[aria-checked="${state}"]`).waitFor()
```

Comment:
I would remove this wait for. Generally we shouldn't have to be waiting for much and let playwright locators do their thing.

Reply:
without this wait you can't write a test like

```ts
const { getState, setState } = useToggle(worksQueue)
const currentState = await getState()
await setState(!currentState) // this is where the `waitFor` is currently
const newState = await getState() // without the `waitFor` this `getState` won't be the updated value
```

Reply:
@pleek91 what do you think about moving it into a separate method

```ts
export function useToggle(page: Page | Locator = PAGE): UseToggle {
  return {
    toggle: getToggle(page),
    getState: (toggle: Locator = getToggle(page)) => {
      return isChecked(toggle)
    },
    setState: async (state, toggle: Locator = getToggle(page)) => {
      const current = await isChecked(toggle)

      if (current !== state) {
        toggle.click()
      }
    },
    waitFor: (state, toggle: Locator = getToggle(page)) => {
      return toggle.locator(`[aria-checked="${state}"]`).waitFor()
    },
  }
}
```